### DISCUSSION TITLE
How to modify the TenantId name

### DISCUSSION THREAD

**Discussion by NickCc6** [abpframework/abp#19744] [Q&A]
hi，What should I do when I need to modify the TenantId to tent_id when using IMultiTenant
1. I tried [Column ("tenant_id")], but it didn't work
2. I tried setting up IEntityTypeConfiguration: builder Property ("TenantId") HasColumnName ("tenant_id"); But it's invalid
Is there any good way for me to solve this problem?

![image](https://github.com/abpframework/abp/assets/42535452/1c88545c-b2bc-4cca-9b4e-9c71ee2dd0c1)

**Comment by maliming** [comment]
> I tried setting up IEntityTypeConfiguration: builder Property ("TenantId") HasColumnName ("tenant_id"); But it's invalid

I think this should work. Can you share your code?

https://github.com/abpframework/abp/blob/dev/framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/EntityFrameworkCore/Modeling/AbpEntityTypeBuilderExtensions.cs#L280-L282

**Reply by NickCc6** [reply]
entity
![image](https://github.com/abpframework/abp/assets/42535452/41f4842f-2e7d-4a72-868e-42a287f2e606)
entityTypeconfig
![image](https://github.com/abpframework/abp/assets/42535452/8e2fcbd7-9686-45b6-94d6-996e0ef9a42a)
DbContext
![image](https://github.com/abpframework/abp/assets/42535452/abe8077b-97aa-4255-a370-cd244d5ce9e2)
Repository
![image](https://github.com/abpframework/abp/assets/42535452/e3686426-fc3d-4841-b2f8-1c382c2ef2d7)
efcore sql
`SET @__ef_filter__p_0 = FALSE;

SELECT `s`.`sys_user_id`, `s`.`birth_date`, `s`.`creator_id`, `s`.`creator_name`, `s`.`creator_time`, `s`.`email`, `s`.`head_img`, `s`.`login_account`, `s`.`login_pwd`, `s`.`mobile`, `s`.`nick_name`, `s`.`real_name`, `s`.`TenantId`, `s`.`update_id`, `s`.`update_name`, `s`.`update_time`, `s`.`user_status`
FROM `sys_user` AS `s`
WHERE @__ef_filter__p_0 OR (`s`.`TenantId` IS NULL)

This LINQ query is being executed in split-query mode, and the SQL shown is for the first query to be executed. Additional queries may also be executed depending on the results of the first query.`

**Reply by NickCc6** [reply]
I found that when executing EntityTypeConfiguration, TenantId=tenant_id
I think ABP executed TryConfigureMultiTenant after that, rewriting the ColumnName of TeamId
But I don't know how to solve this problem

**Reply by maliming** [reply]
Try to move the order of method calls.

```
base.OrModelCreating(modelBuilder):
modelBuilder.ApplyConfigurationsFromAssembly (assembly: typeof (EntitylapHelp). Assembly) :
```

**Reply by NickCc6** [reply]
This is good, thank you for your help

**Reply by NickCc6** [reply]
hi，
Yesterday's issue was resolved, but I found that switching: the filtering for multiple tenants has failed

 base.OnModelCreating(modelBuilder) in after
![image](https://github.com/abpframework/abp/assets/42535452/3ff2abd2-2a82-46de-b786-3b6d5df57578)
![image](https://github.com/abpframework/abp/assets/42535452/9e21cef3-d0d2-4f85-8433-ab771568e8b1)

 base.OnModelCreating(modelBuilder) in before
![image](https://github.com/abpframework/abp/assets/42535452/bfba3688-0898-472e-a9d9-68e8f565c52f)
![image](https://github.com/abpframework/abp/assets/42535452/4550d250-8e9c-4501-9797-dd99cb4b2b6f)

**Reply by maliming** [reply]
Please change the constant string as well.

https://github.com/abpframework/abp/blob/dev/framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/EntityFrameworkCore/AbpDbContext.cs#L780

https://docs.abp.io/en/abp/latest/Data-Filtering#entityframework-core

**Comment by NickCc6** [comment]
I have tried the following methods to solve my problem, but I am not sure if it is reasonable

1.If using IEntityTypeConfiguration to configure table field names, OnModelCreating must be
   `ModelBuilder ApplyConstructionsFromAssembly (typeof (EntityMapHelp) Assembly;`
   `Base OnModelCreating (modelBuilder);`
Otherwise, EFCore Global Query Filters will not be executed

2.  AbpDbContext Set TenantId to TenantId , so override ConfigureBaseProperties to
Set TenantId to tenant_id

MyDbContext

```
public class MyDbContext : AbpDbContext<MyDbContext>
{
    public MyDbContext(DbContextOptions<MyDbContext> options) : base(options)
    {
    }

    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    {
        base.OnConfiguring(optionsBuilder);
    }

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder.ApplyConfigurationsFromAssembly(typeof(EntityMapHelp).Assembly);
        base.OnModelCreating(modelBuilder);
    }

    protected override void ConfigureBaseProperties<TEntity>(ModelBuilder modelBuilder, IMutableEntityType mutableEntityType)
    {
        base.ConfigureBaseProperties<TEntity>(modelBuilder, mutableEntityType);
        var b = modelBuilder.Entity<TEntity>();
        if (b.Metadata.ClrType.IsAssignableTo<IMultiTenant>())
        {
            b.Property(nameof(IMultiTenant.TenantId))
                .IsRequired(false)
                .HasColumnName("tenant_id");
        }
    }

    protected override Expression<Func<TEntity, bool>>? CreateFilterExpression<TEntity>()
    {
        var expression = base.CreateFilterExpression<TEntity>();
        return expression;
    }
}
```
